crypto/serializer/serializer_pkey.c

```diff
@@ -150,7 +150,8 @@ static int serializer_EVP_PKEY_to_bio(OSSL_SERIALIZER_CTX *ctx, BIO *out)
     if (ctx->ser == NULL)
         return 0;
 
-    if (ctx->ser->serialize_object == NULL) {
+    if (ctx->ser->serialize_object == NULL
+        || OSSL_SERIALIZER_provider(ctx->ser) != EVP_KEYMGMT_provider(keymgmt)) {
```

Comment:
Side note: I provided the same patch to @romen recently in relation to #12291.

Reply:
Matt's patch somehow broke the tests for EC in FIPS mode in #12291 that are currently working.

I wish we could merge 12291 before fixing things that break other things!

Reply:
So that implies that the export is not working correctly for some edge case in ec and will need to be fixed.
You may notice that this PR is 60 PR's older than your one :) and it  is actually green.

Reply:
I am not opposed to merge this of course, but I would answer that we have been merging a lot of functionality without adding tests to make sure everything added is covered though.

That one is not green because it only adds tests that should have been showstoppers for previous PRs that got already merged!

Reply:
Are you going to tell me that after your tests work that we have covered everything? :)

Reply:
Obviously 😂

OK, I am rereading my previous comment and it can indeed be read way more aggressively than I intended it to be!

Reply:
Thanks for adding extra tests, I cant always think of every possible scenario..